crates/router/src/bin/collection_runner.rs

```diff
@@ -0,0 +1,152 @@
+
+    if let Some(auth_type) = inner_map.get(&connector_name.to_string()) {
+        match auth_type {
+            ConnectorAuthType::HeaderKey { api_key } => {
+                newman_command
+                    .arg("--env-var")
+                    .arg(format!("connector_api_key={}", trim(api_key)));
+            }
+            ConnectorAuthType::BodyKey { api_key, key1 } => {
+                newman_command
+                    .arg("--env-var")
+                    .arg(format!("connector_api_key={}", trim(api_key)))
+                    .arg("--env-var")
+                    .arg(format!("connector_key1={}", trim(key1)));
+            }
+            ConnectorAuthType::SignatureKey {
+                api_key,
+                key1,
+                api_secret,
+            } => {
+                newman_command
+                    .arg("--env-var")
+                    .arg(format!("connector_api_key={}", trim(api_key)))
+                    .arg("--env-var")
+                    .arg(format!("connector_key1={}", trim(key1)))
+                    .arg("--env-var")
+                    .arg(format!("connector_api_secret={}", trim(api_secret)));
+            }
+            ConnectorAuthType::MultiAuthKey {
+                api_key,
+                key1,
+                key2,
+                api_secret,
+            } => {
+                newman_command
+                    .arg("--env-var")
+                    .arg(format!("connector_api_key={}", trim(api_key)))
+                    .arg("--env-var")
+                    .arg(format!("connector_key1={}", trim(key1)))
+                    .arg("--env-var")
+                    .arg(format!("connector_key2={}", trim(key2)))
+                    .arg("--env-var")
+                    .arg(format!("connector_api_secret={}", trim(api_secret)));
+            }
+            // Handle other ConnectorAuthType variants
+            _ => {
+                println!("Invalid authentication type.");
+            }
+        }
+    } else {
+        println!("Connector not found.");
+    }
+
+    // Add additional environment variables if present
+    if let Ok(gateway_merchant_id) = env::var("GATEWAY_MERCHANT_ID") {
+        newman_command
+            .arg("--env-var")
+            .arg(format!("gateway_merchant_id={}", gateway_merchant_id));
+    }
+
+    if let Ok(gpay_certificate) = env::var("GPAY_CERTIFICATE") {
+        newman_command
+            .arg("--env-var")
+            .arg(format!("certificate={}", gpay_certificate));
+    }
+
+    if let Ok(gpay_certificate_keys) = env::var("GPAY_CERTIFICATE_KEYS") {
+        newman_command
+            .arg("--env-var")
+            .arg(format!("certificate_keys={}", gpay_certificate_keys));
+    }
+
+    newman_command.arg("--delay-request").arg("5");
+
+    // Execute the newman command
+    let output = newman_command.spawn().and_then(|child| {
+        let result = child.wait_with_output()?;
+        Ok(result)
+    });
```

Comment:
Consider just calling `wait()` on the child. This should anyway print out stdout and stderr (in near realtime), since the child process inherits the stdout and stderr from the parent process because of the `spawn()`.

You should then be able to remove the step printing out the output manually, and only assert on the exit code.